### ISSUE TITLE
[miio] OFFLINE (COMMUNICATION_ERROR): maximum number of DatagramSockets reached

### ISSUE THREAD

**Issue by hudarsono** [openhab/openhab-addons#5784]
## Expected Behavior
Adding a new supported device by entering IP address and token should works.

## Current Behavior
It resulted in  OFFLINE (COMMUNICATION_ERROR): maximum number of DatagramSockets reached
Log :
```
2019-07-01 16:41:10.142 [hingStatusInfoChangedEvent] - 'miio:basic:6e3324e1' changed from UNINITIALIZED to INITIALIZING

2019-07-01 16:41:10.158 [hingStatusInfoChangedEvent] - 'miio:basic:6e3324e1' changed from INITIALIZING to OFFLINE

==> /var/log/openhab2/openhab.log <==

2019-07-01 16:41:11.143 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 

java.lang.NullPointerException: null

	at org.openhab.binding.miio.internal.Utils.hexStringToByteArray(Utils.java:40) ~[?:?]

	at org.openhab.binding.miio.handler.MiIoBasicHandler.initializeData(MiIoBasicHandler.java:237) ~[?:?]

	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:?]

	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) ~[?:?]

	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) ~[?:?]

	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]

	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]

	at java.lang.Thread.run(Thread.java:748) [?:?]

==> /var/log/openhab2/events.log <==

2019-07-01 16:41:20.153 [hingStatusInfoChangedEvent] - 'miio:basic:6e3324e1' changed from OFFLINE to OFFLINE (COMMUNICATION_ERROR): maximum number of DatagramSockets reached
```

## Steps to Reproduce (for Bugs)
1. Open paper UI, choose "manually add things"
2. Entering the IP address and token of the device. The token were retrieved from MiToolkit.
3. The device in question is yeelight ceiling lamp v1

## Your Environment
* Version used: openhabian v2.4.0 and xiaomi mi io binding v2.4.0

**Comment by marcelrv** [comment]
Will be closed with #7255

**Comment by marcelrv** [comment]
This is closed with #7255 as connections are now re-used instead of opened.
